src/providers/appium.ts

```diff
@@ -206,6 +203,57 @@ export function getAppBundleId(path: string): Promise<string> {
   });
 }
 
+export async function getConnectedIOSDeviceUDID(): Promise<string> {
+  try {
+    const { stdout } = await execPromise(`xcrun xctrace list devices`);
+
+    const iphoneDevices = stdout
+      .split("\n")
+      .filter((line) => line.includes("iPhone"));
+
+    const realDevices = iphoneDevices.filter(
+      (line) => !line.includes("Simulator"),
+    );
+
+    if (realDevices.length === 0) {
+      throw new Error(
+        `No connected iPhone detected. Please ensure your device is connected and try again.`,
+      );
+    }
+
+    const deviceLine = realDevices[0];
```

Comment:
better to comment above how these lines look like in console. 
if regex breaks for some scenario, we would know what format current code supports